### DISCUSSION TITLE
Multiple colors in text

### DISCUSSION THREAD

**Discussion by johanohly** [python-pillow/Pillow#5134] [Q&A] [answered]
I don't know how, but I really want multiple colors in one text box. I don't *think* it's possible but I want to hear opinions. And if not, how I could do it myself.

**Comment by hugovk** [comment, marked as answer]
You can draw some characters in one colour, and use `textsize` to find out how wide that part was, and draw the next part in another colour with that width offset.

https://pillow.readthedocs.io/en/stable/reference/ImageFont.html

**Reply by nulano** [reply]
This is the perfect use case for the `textlength` / `getlength` functions added in 8.0.0.

**Reply by johanohly** [reply]
Ohh okay, didn't know those. Thanks!

**Comment by radarhere** [comment]
@JohnnyJTH if you happy with this, could you mark this as answered?

**Reply by johanohly** [reply]
Yes I forgot this thanks

**Reply by johanohly** [reply]
Wait, I cannot seem to close it or delete it. How do I do this?

**Reply by radarhere** [reply]
I haven't done it before myself.

https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer
<img width="550" alt="answer" src="https://user-images.githubusercontent.com/3112309/113473783-75ba4600-94b7-11eb-9d2c-1678cf500ef0.png">
